src/sentry/snuba/metrics/datasource.py

```diff
@@ -272,26 +272,29 @@ def get_series(projects: Sequence[Project], query: QueryDefinition) -> dict:
                 "multi-field select with order by clause queries"
             )
 
-        # This query contains an order by clause, and so we are only interested in the
-        # "totals" query
-        initial_snuba_query = next(iter(snuba_queries.values()))["totals"]
-
-        initial_query_results = raw_snql_query(
-            initial_snuba_query, use_cache=False, referrer="api.metrics.totals.initial_query"
-        )
-
-        # We no longer want the order by in the 2nd query because we already have the order of
-        # the group by tags from the first query so we basically remove the order by columns,
-        # and reset the query fields to the original fields because in the second query,
-        # we want to query for all the metrics in the request api call
-        query.orderby = None
-        query.fields = original_query_fields
+        try:
+            # This query contains an order by clause, and so we are only interested in the
+            # "totals" query
+            initial_snuba_query = next(iter(snuba_queries.values()))["totals"]
 
-        snuba_queries = SnubaQueryBuilder(projects, query).get_snuba_queries()
+            initial_query_results = raw_snql_query(
+                initial_snuba_query, use_cache=False, referrer="api.metrics.totals.initial_query"
+            ).get("data")
+        except StopIteration:
+            initial_query_results = {}
 
         # If we do not get any results from the first query, then there is no point in making
```

Comment:
Under which condition will you have nothing in `snuba_queries`? That condition might be worth documenting here.

Reply:
Yeah you are right.. I will add doc and a test case for this